format/OSITrace.py

```diff
@@ -190,17 +190,20 @@ def get_messages_in_index_range(self, begin, end):
                 yield message
 
         elif self.format_type is None:
-            self.scenario_file.seek(0)
-            serialized_message = self.scenario_file.read()
-            INT_LENGTH = len(struct.pack("<L", 0))
-            message_length = 0
+            message_sequence_len = abs_last_offset - abs_first_offset
+            serialized_messages_extract = self.scenario_file.read(message_sequence_len)
+
+            for rel_index, rel_message_offset in enumerate(rel_message_offsets):
+                rel_begin = rel_message_offset + self._int_length
+                rel_end = (
+                    rel_message_offsets[rel_index + 1] - self._int_length
```

Comment:
`self._int_length` must not be subtracted

Reply:
For some reason I cannot reproduce the bug with more than one message extraction with my osi file. Can you maybe share your osi file if possible? Try to validate your osi file with osi visualizer or the osi validator. 

The other issue is the generator property of python. Generators generate the elements on the fly and throw them away. To prevent it would be creating a deep copy of the generator as an iterator.

Reply:
Maybe an error exceptions needs to be added so that the user knows whats going on. Good point!

Reply:
First of all, I would argue with logic. The first `message_offset` is `0`. Every message offset indicates the first byte of the message length indicator of its following serialized message. Thus, it is the byte right after the previous binary message (first offset excluded). The length indicator offset is already taken into account by the computation of `rel_begin`. By also subtracting `self._int_length` from the next message offset to compute `rel_end`, an incomplete serialized message is read missing the last 4 bytes. The next operation starts at the next `message_offset` entry so that these 4 bytes are skipped and never processed.

The osi-validator suffers from the same problem.

I have no appropriate means for file uploads, but this simple script generates an OSI trace file. It uses the same mechanism to pack the osi messages as `txt2osi.py`. The bug is triggers in the last line. If I apply my proposed fix, it works.
```python
import struct
from osi3.osi_version_pb2 import InterfaceVersion, current_interface_version
from osi3.osi_sensorview_pb2 import SensorView

import OSITrace

filePath = "osi_SensorView_tmp.osi"

sv = SensorView()
sv.version.CopyFrom(InterfaceVersion.DESCRIPTOR.file.GetOptions().Extensions[current_interface_version])
with open(filePath, "wb") as fp:
    for i in range(15):
        sv.timestamp.seconds = 0
        sv.timestamp.nanos = i * 5
        msgBin = sv.SerializeToString()
        filePath = r"_osi_SensorView_tmp.osi"
        fp.write(struct.pack("<L", len(msgBin)))
        fp.write(msgBin)

osiTrace = OSITrace.OSITrace()
osiTrace.from_file(filePath, type_name="SensorView")
msgs = list(osiTrace.get_messages())
```

Reply:
Yes you are right! I checked with different files. But I don't get the "Truncated error message". When decoding there are missing fields when I keep the `self._int_length` subtraction happening see output below:

![image](https://user-images.githubusercontent.com/14346967/96235121-ac98b000-0f9a-11eb-9b0b-4147faab80a8.png)

I will adjust it and will also add a meaningful error message when the message generator reaches the end! Thanks for reporting this issue!